apollo-ios-codegen/Sources/IR/IR+RootFieldBuilder.swift

```diff
@@ -243,17 +216,69 @@ class RootFieldBuilder {
     }
   }
 
+  private func add(
+    _ inlineFragment: CompilationResult.InlineFragment,
+    from selection: CompilationResult.Selection,
+    to target: DirectSelections,
+    atTypePath typeInfo: SelectionSet.TypeInfo
+  ) {
+    guard let scope = scopeCondition(
+      for: inlineFragment,
+      in: typeInfo,
+      deferCondition: inlineFragment.deferCondition
+    ) else {
+      return
+    }
+
+    let inlineSelectionSet = inlineFragment.selectionSet
+
+    switch (typeInfo.scope.matches(scope), inlineFragment.deferCondition) {
+    case let (true, .some(deferCondition)):
+      let irTypeCase = buildInlineFragmentSpread(
+        from: inlineSelectionSet,
+        with: scope,
+        inParentTypePath: typeInfo,
+        deferCondition: .init(deferCondition)
+      )
+      target.mergeIn(irTypeCase)
+
+    case (true, .none):
+      addSelections(from: inlineSelectionSet, to: target, atTypePath: typeInfo)
+
+    case (false, .some):
+      let irTypeCase = buildInlineFragmentSpread(
+        toWrap: selection,
+        with: scope,
+        inParentTypePath: typeInfo
+      )
+      target.mergeIn(irTypeCase)
+
+    case (false, .none):
+      let irTypeCase = buildInlineFragmentSpread(
+        from: inlineSelectionSet,
+        with: scope,
+        inParentTypePath: typeInfo
+      )
+      target.mergeIn(irTypeCase)
+    }
+  }
+
   private func scopeCondition(
     for conditionalSelectionSet: ConditionallyIncludable,
-    in parentTypePath: SelectionSet.TypeInfo
+    in parentTypePath: SelectionSet.TypeInfo,
+    deferCondition: CompilationResult.DeferCondition? = nil
```

Comment:
This needs to be refactored because it's confusing why `deferCondition` is passed into the function but not used in the creation of the `ScopeCondition`; it's only used to determine the parent type. Maybe renaming the parameter name is enough?

Reply:
Changed to a `bool` which implies it's more of a flag and not used to create the scope condition.